src/tarantool_schema.c

```diff
@@ -261,19 +261,25 @@ parse_schema_space_value_value(struct schema_field_value *fld,
 	if (mp_typeof(**tuple) != MP_STR)
 		goto error;
 	const char *sfield = mp_decode_str(tuple, &sfield_len);
-	if (memcmp(sfield, "name", sfield_len) == 0) {
+	if (strncmp_strict(sfield, sfield_len, "name")) {
 		if (mp_typeof(**tuple) != MP_STR)
 			goto error;
 		sfield = mp_decode_str(tuple, &fld->field_name_len);
 		fld->field_name = pemalloc(fld->field_name_len, 1);
 		if (!fld->field_name)
 			goto error;
 		memcpy(fld->field_name, sfield, fld->field_name_len);
-	} else if (memcmp(sfield, "type", sfield_len) == 0) {
+	} else if (strncmp_strict(sfield, sfield_len, "type")) {
 		if (mp_typeof(**tuple) != MP_STR)
 			goto error;
 		sfield = mp_decode_str(tuple, &sfield_len);
 		fld->field_type = field_type_by_name(sfield, sfield_len);
+		if (fld->field_type == field_type_MAX)
+			goto error;
```

Comment:
The same question is as for one of the previous commits: what will going on if a user will use decimal or uuid as a field type in a space format and will not use this space in the client? Does it work now? Will it work after the change?

BTW, it seems a format should be added into a testing space in this commit, not where index parts parsing was changed.

Reply:
Fixed (decimal or uuid)

Reply:
Sorry, but I think that the questions are important to understand your approach and it would be good to have explicit answers. However the main discussion about the approach is in https://github.com/tarantool/tarantool-php/pull/164#discussion_r454719648 now and it seems the approach should be the same as for index part types as well as for format field types.

BTW, we can store a super-new type (and so unknown for the connector) using an old index part format. Need to ensure that we handle an unknown type in the same way in both parsing methods.

Reply:
> BTW, we can store a super-new type (and so unknown for the connector) using an old index part format. Need to ensure that we handle an unknown type in the same way in both parsing methods.

For unknown type "field_type_MAX" will be used.